tests/applications/test_convergence_analysis.py

```diff
@@ -788,85 +792,201 @@ def grids() -> list[pp.Grid, pp.MortarGrid]:
+            face_nodes.indptr[face_number] : face_nodes.indptr[face_number + 1]
+        ]
+
+        node_coordinates = []
+        for node in face_node_indices:
+            node_coordinates.append(sd.nodes[:, node])
+        # Obtain the neighboring cells of the face.
+        neighboring_cells = np.where(sd.cell_faces[face_number].todense() != 0)[
+            1
+        ].tolist()
+
+        def compute_volume(sd, cell, nodes):
+            """Compute the volume of a pyramid."""
+            cc = sd.cell_centers[:, cell].reshape(-1, 1)
+            # Compute the normal distance from the cell center to the face.
+            polygon = np.stack(nodes, axis=1)
+            distance = pp.geometry.distances.points_polygon(
+                p=cc, poly=polygon, tol=1e-8
+            )[0]
+            area = 1 / 3 * distance * sd.face_areas[face_number]
+            return area
+
+        def compute_area(sd, cell, nodes):
+            """Compute the area of a triangle."""
+            cc = sd.cell_centers[:, cell]
+            starting_point = nodes[0]
+            end_point = nodes[1]
+            # Compute the normal distance from the cell center to the face.
+            distance, _ = pp.geometry.distances.points_segments(
+                p=cc, start=starting_point, end=end_point
+            )
+            area = 1 / 2 * distance * sd.face_areas[face_number]
+            return area
+
+        area_local = 0
+        for cell in neighboring_cells:
+            if sd.dim == 3:
+                area_local += compute_volume(sd, cell, node_coordinates)
+            elif sd.dim == 2:
+                area_local += compute_area(sd, cell, node_coordinates)
+        meas[face_number] = area_local
+
+    if parameter_weight is not None:
+        meas *= parameter_weight
+    if not is_scalar:
+        meas = meas.repeat(sd.dim)
+
+    # Obtain numerator and denominator to determine the error.
+    numerator = np.sqrt(np.sum(meas * np.abs(true_array - approx_array) ** 2))
+    denominator = np.sqrt(np.sum(meas * np.abs(true_array) ** 2)) if relative else 1.0
+
+    return numerator / denominator
+
+
 @pytest.mark.parametrize("is_relative", [False, True])
 @pytest.mark.parametrize(
-    "is_sd, is_cc, is_scalar",
+    "is_sd, is_cc, is_scalar, parameter_weight",
     [
-        (True, True, True),  # subdomain scalar cell-centered quantity
-        (True, False, True),  # subdomain scalar face-centered quantity
-        (True, True, False),  # subdomain vector cell-centered quantity
-        (True, False, False),  # subdomain vector face-centered quantity
-        (False, True, True),  # interface scalar cell-centered quantity
-        (False, True, False),  # interface vector cell-centered quantity
+        (True, True, True, None),
+        (True, False, True, None),
+        (True, True, False, None),
+        (True, False, False, None),
+        (False, True, True, None),
+        (False, True, False, 1.2),
```

Comment:
Shouldn't `parameter_weight` here just be `False` and `True`. Looking at the code below, I think this was possibly intendend.